### PR TITLE
Currency: Updated protobuf to add to Currency

### PR DESCRIPTION
**Pull request by muskaan2199**
## Checklist

- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

File: p/protobuf/protobuf_ubi_8.5_v21.5.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+git checkout $PACKAGE_VERSION
+git submodule update --init --recursive
+if ! bazel build :protoc :protobuf; then
+    echo "Build and test fails"
+    exit 2
+else
+    echo "Build and test successful"
+    exit 0
+fi
+cp bazel-bin/protoc /usr/local/bin
```
Comment (nikhil-kalbande) [inline p/protobuf/protobuf_ubi_8.5_v21.5.sh:44]: @MuskaanSheik will this copy work if you have exit before?

File: p/protobuf/protobuf_ubi_8.5_v21.5.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+dnf install -y dnf-plugins-core
+dnf copr enable vbatts/bazel -y
+dnf install -y bazel4
+
+git clone $PACKAGE_URL
+cd $PACKAGE_NAME
+git checkout $PACKAGE_VERSION
+git submodule update --init --recursive
+if ! bazel build :protoc :protobuf; then
+    echo "Build and test fails"
```
Comment (nikhil-kalbande) [inline p/protobuf/protobuf_ubi_8.5_v21.5.sh:38]: @muskaan2199 - Can we split build and test steps?

File: p/protobuf/protobuf_ubi_8.5_v21.5.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+
+PACKAGE_NAME=protobuf
+PACKAGE_VERSION=${1:-v21.5}
+PACKAGE_URL=https://github.com/protocolbuffers/protobuf.git
+
+yum install -y sudo
+sudo yum update -y
```
Comment (nikhil-kalbande) [inline p/protobuf/protobuf_ubi_8.5_v21.5.sh:26]: @muskaan2199 you can removed sudo use inside script

File: p/protobuf/build_info.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "package_name" : "protobuf",
+  "github_url": "https://github.com/protocolbuffers/protobuf",
+  "version": "v21.5",
+  "default_branch": "main",
+  "package_dir": "p/protobuf/",
+  "docker_cmd" :"docker build -t ${package_name}:$PACKAGE_VERSION ${dir}",
+  "build_script": "protobuf_ubi_8.5_v21.5.sh",
+
+  "v21.5" : {
+  "build_script":"protobuf_ubi_8.5_v21.5.sh"
```
Comment (nikhil-kalbande) [inline p/protobuf/build_info.json:11]: @muskaan2199 if we are using same build script across version, can we combine into one entry. 
 you can just mention different one only in regex block and common one can go in default section

**Review by nikhil-kalbande — CHANGES_REQUESTED** [review]

File: p/protobuf/protobuf_ubi_8.5_v21.5.sh (hunk 2 above)
Comment (muskaan2199) [reply p/protobuf/protobuf_ubi_8.5_v21.5.sh:37]: @nikhil-kalbande there's no specific test command in the repository

File: p/protobuf/protobuf_ubi_8.5_v21.5.sh (hunk 1 above)
Comment (muskaan2199) [reply p/protobuf/protobuf_ubi_8.5_v21.5.sh]: I've removed that. It's not needed  for the build

**Comment by muskaan2199** [comment]
Attaching the build logs

>Resolving deltas: 100% (86682/86682), done.
Note: switching to 'v21.5'.

>You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

>If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  >git switch -c <new-branch-name>

>Or undo this operation with:

  >git switch -

>Turn off this advice by setting config variable advice.detachedHead to false

>HEAD is now at ab8403459 Update changelog
Submodule 'third_party/benchmark' (https://github.com/google/benchmark.git) registered for path 'third_party/benchmark'
Submodule 'third_party/googletest' (https://github.com/google/googletest.git) registered for path 'third_party/googletest'
Cloning into '/protobuf/third_party/benchmark'...
Cloning into '/protobuf/third_party/googletest'...
Submodule path 'third_party/benchmark': checked out '5b7683f49e1e9223cf9927b24f6fd3d6bd82e3f8'
Submodule path 'third_party/googletest': checked out '5ec7f0c4a113e2f18ac2c6cc7df51ad6afc24081'
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
DEBUG: /root/.cache/bazel/_bazel_root/c10d995b71ee2cb34bd3247d67c66df9/external/rules_jvm_external/coursier.bzl:445:22: The inputs to maven_install.json have changed, but the lock file has not been regenerated. Consider running 'bazel run @unpinned_maven//:pin'
INFO: Analyzed 2 targets (21 packages loaded, 529 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 47.097s, Critical Path: 7.51s
INFO: 193 processes: 6 internal, 187 processwrapper-sandbox.
INFO: Build completed successfully, 193 total actions
Build and test successful

**Comment by nikhil-kalbande** [comment]
> Attaching the build logs
> 
> > Resolving deltas: 100% (86682/86682), done.
> > Note: switching to 'v21.5'.
> 
> > You are in 'detached HEAD' state. You can look around, make experimental
> > changes and commit them, and you can discard any commits you make in this
> > state without impacting any branches by switching back to a branch.
> 
> > If you want to create a new branch to retain commits you create, you may
> > do so (now or later) by using -c with the switch command. Example:
> 
> > git switch -c
> 
> > Or undo this operation with:
> 
> > git switch -
> 
> > Turn off this advice by setting config variable advice.detachedHead to false
> 
> > HEAD is now at ab8403459 Update changelog
> > Submodule 'third_party/benchmark' (https://github.com/google/benchmark.git) registered for path 'third_party/benchmark'
> > Submodule 'third_party/googletest' (https://github.com/google/googletest.git) registered for path 'third_party/googletest'
> > Cloning into '/protobuf/third_party/benchmark'...
> > Cloning into '/protobuf/third_party/googletest'...
> > Submodule path 'third_party/benchmark': checked out '5b7683f49e1e9223cf9927b24f6fd3d6bd82e3f8'
> > Submodule path 'third_party/googletest': checked out '5ec7f0c4a113e2f18ac2c6cc7df51ad6afc24081'
> > Extracting Bazel installation...
> > Starting local Bazel server and connecting to it...
> > DEBUG: /root/.cache/bazel/_bazel_root/c10d995b71ee2cb34bd3247d67c66df9/external/rules_jvm_external/coursier.bzl:445:22: The inputs to maven_install.json have changed, but the lock file has not been regenerated. Consider running 'bazel run @unpinned_maven//:pin'
> > INFO: Analyzed 2 targets (21 packages loaded, 529 targets configured).
> > INFO: Found 2 targets...
> > INFO: Elapsed time: 47.097s, Critical Path: 7.51s
> > INFO: 193 processes: 6 internal, 187 processwrapper-sandbox.
> > INFO: Build completed successfully, 193 total actions
> > Build and test successful

@muskaan2199 I see this is been run with root user. Can you check and upload logs with normal user + sudo permission.

**Comment by muskaan2199** [comment]
@muskaan2199 I see this is been run with root user. Can you check and upload logs with normal user + sudo permission.
In reference to the above comment added the logs with normal user + sudo permission


>>Complete!
Cloning into 'protobuf'...
remote: Enumerating objects: 122458, done.
remote: Counting objects: 100% (218/218), done.
remote: Compressing objects: 100% (125/125), done.
remote: Total 122458 (delta 98), reused 152 (delta 74), pack-reused 122240
Receiving objects: 100% (122458/122458), 115.28 MiB | 28.51 MiB/s, done.
Resolving deltas: 100% (86904/86904), done.
Note: switching to 'v21.5'.

>>You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

>>If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

>>  git switch -c <new-branch-name>

>>Or undo this operation with:

>>  git switch -

>>Turn off this advice by setting config variable advice.detachedHead to false

>>HEAD is now at ab8403459 Update changelog
Submodule 'third_party/benchmark' (https://github.com/google/benchmark.git) registered for path 'third_party/benchmark'
Submodule 'third_party/googletest' (https://github.com/google/googletest.git) registered for path 'third_party/googletest'
Cloning into '/home/user1/protobuf/third_party/benchmark'...
Cloning into '/home/user1/protobuf/third_party/googletest'...
Submodule path 'third_party/benchmark': checked out '5b7683f49e1e9223cf9927b24f6fd3d6bd82e3f8'
Submodule path 'third_party/googletest': checked out '5ec7f0c4a113e2f18ac2c6cc7df51ad6afc24081'
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
DEBUG: /root/.cache/bazel/_bazel_root/2985c12044b1e1fbfb78f339707d9d26/external/rules_jvm_external/coursier.bzl:445:22: The inputs to maven_install.json have changed, but the lock file has not been regenerated. Consider running 'bazel run @unpinned_maven//:pin'
INFO: Analyzed 2 targets (21 packages loaded, 529 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 47.811s, Critical Path: 7.86s
INFO: 193 processes: 6 internal, 187 processwrapper-sandbox.
INFO: Build completed successfully, 193 total actions
Build and test successful

**Pull request closed without merging**